mx.py

```diff
@@ -7089,7 +7126,14 @@ def _returncode(status):
             return _returncode(status)
         remaining = end - time.time()
         if remaining <= 0:
-            abort('Process timed out after {0} seconds: {1}'.format(timeout, ' '.join(args)))
+            msg = 'Process timed out after {0} seconds: {1}'.format(timeout, ' '.join(args))
+            if nonZeroIsFatal:
+                abort(msg)
+            else:
+                log(msg)
+                for p, _ in _currentSubprocesses:
+                    _kill_process_group(p.pid, signal.SIGKILL)
+                return 1
```

Comment:
You are right. I don't think it matters in my use case but it might be important to others.